backend/compact-connect/lambdas/python/search/opensearch_client.py

```diff
@@ -0,0 +1,127 @@
+import time
+
+import boto3
+from cc_common.config import config, logger
+from cc_common.exceptions import CCInternalException
+from opensearchpy import AWSV4SignerAuth, OpenSearch, RequestsHttpConnection
+from opensearchpy.exceptions import ConnectionTimeout, TransportError
+
+# Retry configuration for bulk indexing
+MAX_RETRY_ATTEMPTS = 5
+INITIAL_BACKOFF_SECONDS = 1
+MAX_BACKOFF_SECONDS = 32
+
+
+class OpenSearchClient:
+    def __init__(self):
+        lambda_credentials = boto3.Session().get_credentials()
+        auth = AWSV4SignerAuth(credentials=lambda_credentials, region=config.environment_region, service='es')
+        self._client = OpenSearch(
+            hosts=[{'host': config.opensearch_host_endpoint, 'port': 443}],
+            http_auth=auth,
+            use_ssl=True,
+            verify_certs=True,
+            connection_class=RequestsHttpConnection,
+            pool_maxsize=20,
+        )
+
+    def create_index(self, index_name: str, index_mapping: dict) -> None:
+        self._client.indices.create(index=index_name, body=index_mapping)
```

Comment:
You may want to consider employing an index name alias, to provide a layer of abstraction that will free us for operations like re-indexing data, to a new mapping. Something like:
Index name: `compact_aslp_providers_v1`
Index alias: `compact_aslp_providers`

Queries use the alias, so if we need to adopt a new mapping, this leaves us free to:
- Create a new index, `compact_aslp_providers_v2`
- Leave v1 up in blue-green style
- Update `compact_aslp_providers` alias to add v2 and remove v1
- Delete v1 index